engine/src/test/java/org/kigalisim/validate/LiveTests.java

```diff
@@ -405,45 +405,152 @@ public void testBasicReplace() throws IOException {
+    assertNotNull(program, "Program should not be null");
+
+    // Get the scenario
+    String scenarioName = "Sim";
+    ParsedScenario scenario = program.getScenario(scenarioName);
+    assertNotNull(scenario, "Scenario should exist");
+
+    // Get start and end years from scenario
+    int startYear = scenario.getStartYear();
+    int endYear = scenario.getEndYear();
+    assertEquals(1, startYear, "Start year should be 1");
+    assertEquals(10, endYear, "End year should be 10");
+
+    // Create engine and machine
+    Engine engine = new SingleThreadEngine(startYear, endYear);
+    PushDownMachine machine = new SingleThreadPushDownMachine(engine);
+
+    // Execute the default policy
+    ParsedPolicy defaultPolicy = program.getPolicy("default");
+    assertNotNull(defaultPolicy, "Default policy should exist");
+    executePolicy(defaultPolicy, machine);
+
+    // Execute other policies in the scenario
+    for (String policyName : scenario.getPolicies()) {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Verify results for year 1 (before replacement)
+    List<EngineResult> results = engine.getResults();
+
+    // Check Sub A in year 1
+    EngineResult resultSubA1 = findResult(results, "Test", "Sub A", 1);
+    assertNotNull(resultSubA1, "Should have result for Test/Sub A in year 1");
+    assertEquals(100000.0, resultSubA1.getManufacture().getValue().doubleValue(), 0.0001,
+        "Sub A Manufacture should be 100000 kg in year 1");
+
+    // Check Sub A GHG consumption in year 1
+    assertNotNull(resultSubA1.getGhgConsumption(), "Sub A should have GHG consumption in year 1");
+    assertEquals(100000.0, resultSubA1.getGhgConsumption().getValue().doubleValue(), 0.0001,
+        "Sub A GHG consumption should be 100000 tCO2e in year 1");
+    assertEquals("tCO2e", resultSubA1.getGhgConsumption().getUnits(),
+        "Sub A GHG consumption units should be tCO2e in year 1");
+
+    // Check Sub B in year 1
+    EngineResult resultSubB1 = findResult(results, "Test", "Sub B", 1);
+    assertNotNull(resultSubB1, "Should have result for Test/Sub B in year 1");
+    assertEquals(0.0, resultSubB1.getManufacture().getValue().doubleValue(), 0.0001,
+        "Sub B Manufacture should be 0 kg in year 1");
+
+    // Check Sub B GHG consumption in year 1
+    assertNotNull(resultSubB1.getGhgConsumption(), "Sub B should have GHG consumption in year 1");
+    assertEquals(0.0, resultSubB1.getGhgConsumption().getValue().doubleValue(), 0.0001,
+        "Sub B GHG consumption should be 0 tCO2e in year 1");
+    assertEquals("tCO2e", resultSubB1.getGhgConsumption().getUnits(),
+        "Sub B GHG consumption units should be tCO2e in year 1");
+
+    // Move to year 10 and check results
+    for (int year = 2; year <= 10; year++) {
+      engine.incrementYear();  // This may be trouble
```

Comment:
This isn't actually running calculations. Please transition `LiveTests` to use `KigaliSimFacade`.